lib/howl/buffer.moon

```diff
@@ -14,22 +14,46 @@ ffi = require 'ffi'
 append = table.insert
 min = math.min
 
-buffer_titles = setmetatable {}, __mode: 'v'
-title_counters = {}
-
-file_title = (file) ->
-  title = file.basename
-  while buffer_titles[title]
-    file = file.parent
-    return title if not file
-    title = file.basename .. File.separator .. title
-
-  title
-
-title_counter = (title) ->
-  title_counters[title] = 1 if not title_counters[title]
-  title_counters[title] += 1
-  title_counters[title]
+file_references = {}
```

Comment:
I don't see how this gets cleaned up. The ref count handling does nothing to remove entries e.g. when a buffer is closed.